dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/CheckAccount.cshtml.cs

```diff
@@ -0,0 +1,122 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeacherIdentity.AuthServer.Helpers;
+using TeacherIdentity.AuthServer.Models;
+using TeacherIdentity.AuthServer.Services.UserVerification;
+
+namespace TeacherIdentity.AuthServer.Pages.SignIn.Register;
+
+public class CheckAccount : PageModel
+{
+    private readonly IIdentityLinkGenerator _linkGenerator;
+    private readonly TeacherIdentityServerDbContext _dbContext;
+    private readonly IUserVerificationService _userVerificationService;
+    private readonly IClock _clock;
+
+    public CheckAccount(
+        IIdentityLinkGenerator linkGenerator,
+        TeacherIdentityServerDbContext dbContext,
+        IUserVerificationService userVerificationService, IClock clock)
+    {
+        _linkGenerator = linkGenerator;
+        _dbContext = dbContext;
+        _userVerificationService = userVerificationService;
+        _clock = clock;
+    }
+
+    public string? Email => HttpContext.GetAuthenticationState().EmailAddress;
+    public string? ExistingEmail => HttpContext.GetAuthenticationState().ExistingAccountEmail;
+
+    [BindProperty]
+    [Display(Name = " ")]
+    [Required(ErrorMessage = "Content TBD")]
+    public bool? IsUsersAccount { get; set; }
+
+    public void OnGet()
+    {
+    }
+
+    public async Task<IActionResult> OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        var authenticationState = HttpContext.GetAuthenticationState();
+        authenticationState.OnExistingAccountConfirmed((bool)IsUsersAccount!);
+
+        if (IsUsersAccount == true)
+        {
+            await TryGenerateEmailPinForEmail(ExistingEmail!);
```

Comment:
I think it's worth checking the result of this to make sure it succeeded and throwing if not.